fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PruneOlapScanPartition.java

```diff
@@ -0,0 +1,166 @@
+            // TODO: 1. support grammar: SELECT * FROM tbl PARTITION(p1,p2)
+            //       2. support list partition
+            if (partitionColumnNameSet.isEmpty() || !partitionInfo.getType().equals(PartitionType.RANGE)) {
+                return ctx.root;
+            }
+            // TODO: Process all partition column for now, better to process required column only.
+            Map<String, ColumnRange> columnNameToRange = Maps.newHashMap();
+            for (String colName : partitionColumnNameSet) {
+                ColumnRange columnRange = createColumnRange(colName, expressionList);
+                columnNameToRange.put(colName, columnRange);
+            }
+
+            Map<Long, PartitionItem> keyItemMap = partitionInfo.getIdToItem(false);
+            PartitionPruner partitionPruner = new RangePartitionPrunerV2(keyItemMap,
+                    partitionInfo.getPartitionColumns(), columnNameToRange);
+            Collection<Long> selectedPartitionId =  Utils.execWithReturnVal(partitionPruner::prune);
+            scan.getSelectedPartitionIds().retainAll(selectedPartitionId);
+            return ctx.root;
+        }).toRule(RuleType.PARTITION_PRUNE);
+    }
+
+    private ColumnRange createColumnRange(String colName, List<Expression> expressionList) {
+        ColumnRange result = ColumnRange.create();
+        for (Expression expression : expressionList) {
+            Expression leftMostChild = expression.leftMostNode();
+            if (!(leftMostChild instanceof SlotReference)) {
+                continue;
+            }
+            SlotReference slotRef = (SlotReference) leftMostChild;
+            if (!slotRef.boundToColumn(colName)) {
+                continue;
+            }
+            if (expression instanceof Or) {
+                ColumnRanges ranges = exprToRanges(expression, colName);
+                switch (ranges.type) {
+                    case IS_NULL:
+                        result.setHasConjunctiveIsNull(true);
+                        break;
+                    case CONVERT_SUCCESS:
+                        result.intersect(ranges.ranges);
+                        break;
+                    case CONVERT_FAILURE:
+                    default:
+                        break;
+                }
+            } else {
+                ColumnRanges ranges = exprToRanges(expression, colName);
+                switch (ranges.type) {
+                    case IS_NULL:
+                        result.setHasConjunctiveIsNull(true);
+                        break;
+                    case CONVERT_SUCCESS:
+                        result.intersect(ranges.ranges);
+                        break;
+                    case CONVERT_FAILURE:
+                    default:
+                        break;
+                }
+            }
+        }
+        return result;
+    }
+
+    private ColumnRanges exprToRanges(Expression expression, String colName) {
+        // TODO: process in/is null expression
```

Comment:
is these needed by TPC-H?

Reply:
It's determined on how the the partition is set for tpch table.